components/datetime/src/provider/gregory.rs

```diff
@@ -31,8 +32,17 @@ icu_provider::impl_data_marker_no_lifetime!(
 pub struct DatePatternsV1 {
     pub date: patterns::LengthPatternsV1,
 
+    /// The time patterns are used if no hour cycle preference is set. These patterns also represent
+    /// the default hour cycle preference for the locale given the length.
     pub time: patterns::LengthPatternsV1,
 
+    /// The flexible hour cycle that is used in the `time`.
+    pub preferred_hour_cycle: pattern::FlexibleHourCycle,
+
+    /// Users can override the hour cycle with a preference. These patterns contain the
+    /// non-preferred hour cycles.
+    pub time_with_alt_hour_cycle: patterns::LengthPatternsV1,
```

Comment:
Suggestion: it would make more sense to me if you had a field that was always h11/h12 and the other always h23/h24, and the preferred_hour_cycle lets you pick between them.  So more like

```rust
    pub time_h12: patterns::LengthPatternsV1,
    pub time_h23: patterns::LengthPatternsV1,
    pub preferred_hour_cycle: pattern::FlexibleHourCycle,
```

Reply:
Ah, that is much nicer.